src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -386,6 +513,55 @@ public override void InvokeCallback(bool allowPooling)
 
                 cb(bt, sa, sal, SocketFlags.None, ec);
             }
+
+            internal override unsafe bool TryBatch(SocketAsyncContext context, int id, ref Interop.Sys.IoControlBlock ioControlBlock)
+            {
+                if (IsSync || !TrySetRunning())
+                {
+                    return false;
+                }
+
+                PinHandle = Buffer.Pin();
+
+                ioControlBlock.AioData = (ulong)id;
+                ioControlBlock.AioLioOpcode = Interop.Sys.IoControlBlockFlags.IOCB_CMD_PWRITE;
+                ioControlBlock.AioFildes = (uint)context._socket.DangerousGetHandle().ToInt32();
+                ioControlBlock.AioBuf = (ulong)PinHandle.Pointer;
+                ioControlBlock.AioNbytes = (ulong)Count;
+                ioControlBlock.AioOffset = Offset;
+
+                return true;
+            }
+
+            internal override void HandleBatchEvent(in Interop.Sys.IoEvent ioControlBlock)
+            {
+                PinHandle.Dispose();
+
+                if (ioControlBlock.Res < 0)
+                {
+                    ErrorCode = Map((Interop.Error)(-ioControlBlock.Res));
+
+                    if (CanRetry(ErrorCode))
```

Comment:
It would be good to handle `EINTR` too.
I'm not sure if it can happen, but better be safe than sorry.
In case of `EINTR`, the operation should be tried again.